test/conformance/event/fixtures.h

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2022-2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#ifndef UR_CONFORMANCE_EVENT_FIXTURES_H_INCLUDED
+#define UR_CONFORMANCE_EVENT_FIXTURES_H_INCLUDED
+
+#include <uur/fixtures.h>
+
+namespace uur {
+namespace event {
+
+/**
+ * Test fixture that is intended to be used when testing reference count APIs
+ * (i.e. urEventRelease and urEventRetain). Does not handle destruction of the
+ * event.
+ */
+struct urEventReferenceTest : uur::urQueueTest {
+
+    void CreateEvent() {
+        ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_WRITE_ONLY, size,
+                                         nullptr, &buffer));
+
+        input.assign(count, 42);
+        ASSERT_SUCCESS(urEnqueueMemBufferWrite(
+            queue, buffer, false, 0, size, input.data(), 0, nullptr, &event));
+    }
+
+    void checkEventReferenceCount(uint32_t expected_value) {
+        uint32_t reference_count{};
+        urEventGetInfo(event, ur_event_info_t::UR_EVENT_INFO_REFERENCE_COUNT,
+                       sizeof(uint32_t), &reference_count, nullptr);
+        ASSERT_EQ(reference_count, expected_value);
+    }
+
+    void Cleanup() {
```

Comment:
This can be `TearDown()` - mean you don't have to call it manually

Reply:
I agree, having `CreateEvent()` and `Cleanup()` instead of just doing work in overridden `SetUp()` and `TearDown()` actually adds unnecessary repetition to the test cases.

It also diverges from the patterns in all the other test fixtures which I found jarring when reviewing this code, being different requires extra thought for the reader.

Reply:
Done